### DISCUSSION TITLE
webcal link not https

### DISCUSSION THREAD

**Discussion by Maggi68** [bitfireAT/icsx5#190] [Q&A] [answered]
Hello,
I tried the app for my Test Homepage with a Calendar component at Joomla 4.4, called DPcalendar. There is a option to subscribe to a webcal link for each calendars. See https://j4.gfg-fortbildung.de/index.php/veranstaltungen#year=2023&month=11&day=5&view=month
Open the triangel above the calendar view and select "abbonieren".
 If I copy this link it is like this webcal://j4.gfg-fortbildung.de/index.php?option=com_dpcalendar&task=ical.download&id=44
And it open a dialog to open a calendar app like Thundebird at my Windows PC. If I use a Android phone with installed ICSx5 a dialog window opens and the link will be displyed at the app. But than it is not a webcal link, now it is a http url link: http://j4.gfg-fortbildung.de/index.php?option=com_dpcalendar&task=ical.download&id=44.
But without https only http, so the link doesnot work.
Is their a solution to solve this problem?
Thanks

**Comment by ArnyminerZ** [comment, marked as answer]
webcal is not a valid URL protocol, is kind of a way of stating that the link leads to a  webcal link. Actually, webcal means http, and webcals https. So it should work perfectly.

It looks like the URL you are using expects https, so you can simply use the link https://j4.gfg-fortbildung.de/index.php?option=com_dpcalendar&task=ical.download&id=44. It should work correctly.

**Comment by PanderMusubi** [comment]
Other calendar software such as Thunderbird will try first https for webcal and as a fallback http. Perhaps that is a good way to handle these URLs.

Software that supports webcal is https://en.wikipedia.org/wiki/Webcal#Handlers

My proposal:
- http: use directly, fallback https
- https: use directly
- webcal: use https, fallback http
- webcals: use https, fallback http

This is very similar to how Thunderbird works.

**Reply by ArnyminerZ** [reply]
Hi, the code right now does not make any verification on the availability of the server when using `webcal`(`s`), it simply replaces the protocol with `http`(`s`). Arguably https is always more secure, but in any case, it's kind of responsibility of the server to notify about SSL availability with `webcals`.

In any case, ignoring ICSx⁵, if `webcal://example.com` was ever used (as technically given by the server), most clients would use `http` to make the request. Even though some may try `https` before, it's technically not asked by the user, and I'd even argue that it's doubling the requests we are making to the server, if we try `https` before, and if it doesn't work, try `http`.

So even though I understand your point, I don't see any benefits on using that approach. @sunkup may have other things to say.

**Reply by sunkup** [reply]
I agree with @ArnyminerZ ! I see the idea, but it'd become too much work for little gain. Too much other much more important work atm. Sorry.
